Libraries/PushNotificationIOS/RCTPushNotificationManager.m

```diff
@@ -145,6 +159,25 @@ - (void)handleRemoteNotificationsRegistered:(NSNotification *)notification
   [self sendEventWithName:@"remoteNotificationsRegistered" body:notification.userInfo];
 }
 
+- (void)handleRegisterUserNotificationSettings:(NSNotification *)notification
+{
+  if (self.requestPermissionsResolveBlock == nil) return;
+
+  NSMutableDictionary* notificationTypes = [NSMutableDictionary dictionaryWithObjectsAndKeys:@NO, @"alert", @NO, @"badge", @NO, @"sound", nil];
```

Comment:
I think @javache meant

    @((BOOL)(notificationSettings.types & UIUserNotificationTypeAlert))

So that the value of the NSNumber is 1 or 0 instead of the numeric value of the UIUserNotificationTypeAlert enum.

However, that won't work either, since casting to a BOOL doesn't actually affect the value. I think what you'd actually need to write is

    @(notificationSettings.types & UIUserNotificationTypeAlert > 0)

That's equivalent to 

    [NSNumber numberWithBool:notificationSettings.types & UIUserNotificationTypeAlert]

But we prefer the modern @(...) syntax to [NSNumber numberWithX:...] in our codebase.